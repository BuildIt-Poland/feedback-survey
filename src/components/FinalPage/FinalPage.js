import React from 'react';
import styled from 'styled-components';

import { colorRed } from '../../styles/designTokens';
import { THANK_YOU, ERROR_OCCURED } from '../../constatnts/messages';
import { finalLocationState } from '../../types';
import Page from '../Layout/Page';
import StyledHeading from '../Layout/Heading';
import Paragraph from '../Layout/Paragraph';

const Wrapper = styled(Page)`
  background: url('/image/BinarycodeWithHearts.png') no-repeat;
  background-size: 80%;
`;

const renderFinalMessage = (isSuccessSave, saveErrorMessage) => {
  if (isSuccessSave) {
    return (
      <StyledHeading level="1" textAlign="center">
        {THANK_YOU}
      </StyledHeading>
    );
  }
  return (
    <Paragraph color={colorRed} textAlign="center">
      {`${ERROR_OCCURED}: ${saveErrorMessage}`}
    </Paragraph>
  );
};

const FinalPage = ({ location }) => (
  <Wrapper>{renderFinalMessage(location.state.isSuccessSave, location.state.saveErrorMessage)}</Wrapper>
);

FinalPage.propTypes = {
  location: finalLocationState
};

export default FinalPage;
