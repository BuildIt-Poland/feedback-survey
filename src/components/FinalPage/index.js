import { withRouter } from 'react-router-dom';
import FinalPage from './FinalPage';

export default withRouter(FinalPage);
