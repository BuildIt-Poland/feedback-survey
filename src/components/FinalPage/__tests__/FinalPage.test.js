import React from 'react';
import { create } from 'react-test-renderer';

import FinalPage from '../FinalPage';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('COMPONENT - FinalPage', () => {
  it('renders correctly', () => {
    const location = {
      state: {
        success: true
      }
    };

    const component = create(<FinalPage location={location} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
