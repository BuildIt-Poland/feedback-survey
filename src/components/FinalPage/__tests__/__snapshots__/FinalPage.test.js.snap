// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - FinalPage renders correctly if submit fail 1`] = `
<section
  className="sc-bdVaJa sc-bxivhb hCqsPe"
>
  <p
    className="sc-htpNat jfeDHX"
    color="#E74536"
  >
    Error occured: error occured
  </p>
</section>
`;

exports[`COMPONENT - FinalPage renders correctly if submit success 1`] = `
<section
  className="sc-bdVaJa sc-bxivhb hCqsPe"
>
  <h1
    className="sc-bwzfXH fIpTJf StyledHeading-sc-1rdh4aw-0 iDyXcl"
  >
    Thank you for taking the time to complete this survey.
  </h1>
</section>
`;
